### ISSUE TITLE
[data grid] How do I fix this? - `GridVirtualScrollbar.js:88 Uncaught TypeError: Cannot set properties of null (setting 'scrollTop')`

### ISSUE THREAD

**Issue by swensel** [mui/mui-x#13301]
### The problem in depth

I'm facing an issue and I'm not sure how easy it will be to create a simple reproducible example for this, but I'm wondering if there's any pointers you can give.

I have upgraded the following libraries:

```
"@mui/icons-material": "^5.3.1",
"@mui/material": "^5.4.0",
"@mui/x-data-grid": "^5.5.0",
"@mui/x-data-grid-pro": "^5.12.0",
"@mui/x-date-pickers": "^5.0.14",
"@mui/x-license-pro": "^5.12.0"
```

to these new versions, and that's when I started to notice this issue:

```
"@mui/icons-material": "^5.15.19",
"@mui/material": "^5.15.19",
"@mui/x-data-grid": "^7.5.1",
"@mui/x-data-grid-pro": "^7.5.1",
"@mui/x-date-pickers": "^7.5.1",
"@mui/x-license": "^7.2.1"
```

I have a grid that is using `@mui/x-data-grid-pro`, that also has some custom quick filters. 

Sometimes, after selecting or clearing the quick filters, I get this error:

```
GridVirtualScrollbar.js:88 Uncaught TypeError: Cannot set properties of null (setting 'scrollTop')
    at GridVirtualScrollbar.js:88:1
    at HTMLDivElement.<anonymous> (useEventCallback.js:18:1)
```

![Untitled](https://github.com/mui/mui-x/assets/6386584/6c4cda28-d221-4f44-8722-c7f7c8f1ca79)

That is coming from inside the MUI library:

![image](https://github.com/mui/mui-x/assets/6386584/9cab854a-c567-47c1-a24a-bec558ce343e)

There's no way I can catch this error or anything, due to it being from inside the library.  The grid still loads after, but it looks pretty bad on local, and it will also be logged to the console regardless.

I did find this [scrollTop issue](https://v5-0-6.mui.com/guides/migration-v4/#cannot-read-property-scrolltop-of-null) mentioned in the v4 to v5 migration, which looks like a similar error, but I don't think it is the same case.  I don't see anything mentioned about the specific error in the [v5 to v6 migration](https://mui.com/x/migration/migration-data-grid-v5/) or [v6 to v7 migration](https://mui.com/x/migration/migration-data-grid-v6/) docs.

Any idea how to fix or catch this error?

Thanks.

### Your environment

<details>
  <summary>`npx @mui/envinfo`</summary>

```
  System:
    OS: macOS 14.5
  Binaries:
    Node: 18.20.2 - ~/.nvm/versions/node/v18.20.2/bin/node
    npm: 10.5.0 - ~/.nvm/versions/node/v18.20.2/bin/npm
    pnpm: Not Found
  Browsers:
    Chrome: 125.0.6422.61
    Edge: 121.0.2277.128
    Safari: 17.5
  npmPackages:
    @emotion/react: ^11.7.1 => 11.11.4 
    @emotion/styled: ^11.6.0 => 11.11.5 
    @mui/base:  5.0.0-beta.40 
    @mui/core-downloads-tracker:  5.15.19 
    @mui/icons-material: ^5.15.19 => 5.15.19 
    @mui/material: ^5.15.19 => 5.15.19 
    @mui/private-theming:  5.15.14 
    @mui/styled-engine:  5.15.14 
    @mui/system:  5.15.15 
    @mui/types:  7.2.14 
    @mui/utils:  5.15.14 
    @mui/x-data-grid: ^7.5.1 => 7.5.1 
    @mui/x-data-grid-pro: ^7.5.1 => 7.5.1 
    @mui/x-date-pickers: ^7.5.1 => 7.5.1 
    @mui/x-license: ^7.2.1 => 7.2.1 
    @types/react: ^17.0.1 => 17.0.80 
    react: ^17.0.1 => 17.0.2 
    react-dom: ^17.0.1 => 17.0.2 
    typescript: ^4.8.4 => 4.9.5 
```
</details>

**Search keywords**: scrolltop
**Order ID**: 90881

**Comment by michelengelen** [comment]
Hey @swensel 
Thanks for opening an issue here.

I cannot reproduce the issue you are mentioning.

Is there any customization you did to the data grid?
Even CSS stylings could be important, since this is related to the scrollbar reference not being present although it should be.

**Comment by michelengelen** [comment]
So, the grid is fine until there are things rendered that come later? In your case you do render quickfilters after the initial render, is that correct? That could already be a hint to whats going wrong.

Could you try to replicate using a minimal reconstruction of your implementation? Without that it will be near impossible to find the issue.

**Comment by michelengelen** [comment]
> Hi @michelengelen, the plan is to try to provide a standalone demo. That is just going to take some time as my team has other priorities. Should I file a new issue for that when it is ready, since this was automatically closed? Thanks.

No, you can open a new one once the demo is ready and reference this! 👍🏼

**Comment by theodoremoreland** [comment]
I'm not the original poster, but I did encounter the same error. I created a repo dedicated to recreating and documenting the error here: https://github.com/theodoremoreland/mui-x-data-grid-bug-demo

I used to combination of package version to similar effect. The main branch features older versions as those are similar to what I use at work (where I encountered the error). The other branch uses latest versions for most packages.

I will consider deploying the code and/or creating a new issue tomorrow.

**Comment by theodoremoreland** [comment]
I deployed an example where you can reproduce the bug here: https://theodoremoreland.github.io/mui-x-data-grid-bug-demo/

The conditions to reproduce the bug are very conditional, if not inconsistent. For example, from what I can tell:
- Chrome and Firefox experience different behavior and even display different error messages
- Interacting with the virtual scroll bar via scroll wheel as opposed to mouse click+drag has different effects
- Moving the virtual scroll bar to the absolute bottom of the available space is often required to reproduce bug
- The UI will display the error if running on local, but it will only appear in the dev console if running via deployment build
- The syntax used to render React's root DOM node makes a difference given given "react": "^18.3.1", "react-dom": "^18.3.1", and "react-scripts": "5.0.1"

## The bug

When scrolling to the very bottom of the virtual scrollbar using a scroll wheel then decreasing the number of rows in a Data Grid via clicking a lesser page option or via programmatically deleting rows via some custom UI prompt (such as a Delete button) such that the vertical, virtual scroll bar is no longer needed, the Data Grid will throw the following error on `Chrome`:

```
Cannot set properties of null (setting 'scrollTop')
TypeError: Cannot set properties of null (setting 'scrollTop')
    at http://localhost:3000/static/js/bundle.js:49775:31
    at HTMLDivElement.<anonymous> (http://localhost:3000/static/js/bundle.js:37491:19)
```

The following error on `Firefox`:

```
scrollbar is null
./node_modules/@mui/x-data-grid/components/virtualization/GridVirtualScrollbar.js/GridVirtualScrollbar/onScrollerScroll<@http://localhost:3000/static/js/bundle.js:49775:41
./node_modules/@mui/utils/esm/useEventCallback/useEventCallback.js/useEventCallback/<@http://localhost:3000/static/js/bundle.js:37491:19
```

_NOTE: Reproducing the error via programmatic deletion is inconsistent on Firefox (in my experience). However, the error occurs consistently in Chrome and in both browsers via decreasing page option._

This bug was actually discovered and reported by another person prior, but the thread was closed temporarily: https://github.com/mui/mui-x/issues/13301

### How to recreate

#### Use case 1

_NOTE: Oddly enough, scrolling to the bottom of the grid / virtual scrollbar
by clicking and dragging the scroll bar down does not produce an error nor does scrolling close to the bottom. The error is only produced by scrolling to the absolute bottom of the scrollbar via scroll wheel in this example._

1. Open app in either Chrome of Firefox web browser
2. Click OPEN button to open Drawer component featuring two Data Grids
3. Click Grid #2 tab
4. Scroll to the absolute bottom of the data grid using **scroll wheel** (not by dragging scroll bar down with mouse)
5. Change Rows per page option to 2 instead of 100

If running the code locally, you should see an error popup in the UI. If running a deployed build, you may only see the error in the dev console.

#### Screenshots 1

<img src="https://raw.githubusercontent.com/theodoremoreland/mui-x-data-grid-bug-demo/refs/heads/main/screenshots/1.png" width="600">
<img src="https://raw.githubusercontent.com/theodoremoreland/mui-x-data-grid-bug-demo/refs/heads/main/screenshots/2.png" width="600">
<img src="https://raw.githubusercontent.com/theodoremoreland/mui-x-data-grid-bug-demo/refs/heads/main/screenshots/3.png" width="600">
<img src="https://raw.githubusercontent.com/theodoremoreland/mui-x-data-grid-bug-demo/refs/heads/main/screenshots/4.png" width="600">
<img src="https://raw.githubusercontent.com/theodoremoreland/mui-x-data-grid-bug-demo/refs/heads/main/screenshots/5.png" width="600">

#### Use case 2

_NOTE: Weirdly enough, the following steps produce an error far more often
when using Google Chrome, in my experience. It may take multiple attempts
to get the error to produce in Firefox if at all._

1. Open app in either Chrome of Firefox web browser
2. Click OPEN button to open Drawer component featuring two Data Grids
3. Click Grid #2 tab
4. Scroll to the absolute bottom of the data grid using **scroll wheel** (not by dragging scroll bar down with mouse)
5. Click the DELETE button (top right, just above Data Grid)

If running the code locally, you should see an error popup in the UI. If running a deployed build, you may only see the error in the dev console.

#### Screenshots 2

<img src="https://raw.githubusercontent.com/theodoremoreland/mui-x-data-grid-bug-demo/refs/heads/main/screenshots/1.png" width="600">
<img src="https://raw.githubusercontent.com/theodoremoreland/mui-x-data-grid-bug-demo/refs/heads/main/screenshots/2.png" width="600">
<img src="https://raw.githubusercontent.com/theodoremoreland/mui-x-data-grid-bug-demo/refs/heads/main/screenshots/3.png" width="600">
<img src="https://raw.githubusercontent.com/theodoremoreland/mui-x-data-grid-bug-demo/refs/heads/main/screenshots/6.png" width="600">
<img src="https://raw.githubusercontent.com/theodoremoreland/mui-x-data-grid-bug-demo/refs/heads/main/screenshots/7.png" width="600">

### A potential workaround

For some reason, how you render React's root DOM node can circumvent the error. For example, given "react": "^18.3.1" and "react-dom": "^18.3.1":

This does not work:

```
import React from "react";
import ReactDOM from "react-dom"; // ? This version of ReactDOM causes error
import "./index.css";
import App from "./App";
import reportWebVitals from "./reportWebVitals";

ReactDOM.render(
  <React.StrictMode>
    <App />
  </React.StrictMode>,
  document.getElementById("root") as HTMLElement
);

// If you want to start measuring performance in your app, pass a function
// to log results (for example: reportWebVitals(console.log))
// or send to an analytics endpoint. Learn more: https://bit.ly/CRA-vitals
reportWebVitals();
```

This does:

```
import React from 'react';
import ReactDOM from 'react-dom/client'; // ? This version of ReactDOM does not cause error
import './index.css';
import App from './App';
import reportWebVitals from './reportWebVitals';

const root = ReactDOM.createRoot(
  document.getElementById("root") as HTMLElement
);
root.render(
  <React.StrictMode>
    <App />
  </React.StrictMode>
);

// If you want to start measuring performance in your app, pass a function
// to log results (for example: reportWebVitals(console.log))
// or send to an analytics endpoint. Learn more: https://bit.ly/CRA-vitals
reportWebVitals();
```

**Comment by theodoremoreland** [comment]
I created a new issue because this

> > Hi @michelengelen, the plan is to try to provide a standalone demo. That is just going to take some time as my team has other priorities. Should I file a new issue for that when it is ready, since this was automatically closed? Thanks.
> 
> No, you can open a new one once the demo is ready and reference this! 👍🏼

I created a new issue with demo and reproduce steps here: https://github.com/mui/mui-x/issues/14987
